applications/asset_tracker/src/main.c

```diff
@@ -1878,6 +1905,14 @@ static void lte_handler(const struct lte_lc_evt *const evt)
 	case LTE_LC_EVT_CELL_UPDATE:
 		LOG_INF("LTE cell changed: Cell ID: %d, Tracking area: %d",
 			evt->cell.id, evt->cell.tac);
+#if defined(CONFIG_AGPS_SINGLE_CELL_ONLY)
+		/* Request location based on new network info */
+		if (data_send_enabled() && evt->cell.id != -1) {
+			k_delayed_work_submit_to_queue(&application_work_q,
+						       &send_agps_request_work,
```

Comment:
I think this can lead to pretty frequent requests also in stationary situations.
Also I'm not sure if we actually need the device to know its location in the asset tracker? I'm thinking it's sufficient to report cell ID to the cloud and save the power and data traffic required to retrieve the location, but open to other views

Reply:
The asset tracker certainly doesn't need then location.  Perhaps a flag can be added to the cloud message to indicate if the device wants the location?  I assume that there are some use-cases where a device might want the location?

Reply:
If you view this as an example for customers to base their designs, it is good to include the ability to have the location sent to the device, though I agree, in the narrow case of an unmodified asset tracker, it is not needed.  However, I'm not sure if the map tile on nRF Cloud updates unless asset tracker sends the location back to the cloud.  Seems like in single cell mode, it should, in order to save bandwidth.

Reply:
ticket `IRIS-2637` will handle displaying single-cell location, so no need for the device to send it back to the cloud.

Reply:
I'm sure there are use-cases for this feature, so sounds like a good idea to add a flag to the request to signal if the device wants the response back or just intends to report latest seen cell to the cloud for it to process without returning the location